roachtest: sqlsmith/setup=seed-vec/setting=vec failed

[(roachtest).sqlsmith/setup=seed-vec/setting=vec failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2870055&tab=buildLog) on [release-21.1@5323c44e1c3d3d9ded95a9f20be5ba738cd542ec](https://github.com/cockroachdb/cockroach/commits/5323c44e1c3d3d9ded95a9f20be5ba738cd542ec):

```
The test failed on branch=release-21.1, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/sqlsmith/setup=seed-vec/setting=vec/run_1
	sqlsmith.go:206,sqlsmith.go:237,test_runner.go:767: error: pq: internal error: unexpected error from the vectorized engine: runtime error: index out of range [1] with length 1
		stmt:
		SELECT
			count(*) AS col_424
		FROM
			defaultdb.public.seed_vec@[0] AS tab_114
		GROUP BY
			tab_114._int8, tab_114._date
		HAVING
			true
		ORDER BY
			tab_114._date
		LIMIT
			32:::INT8;

```

<details><summary>More</summary><p>


Artifacts: [/sqlsmith/setup=seed-vec/setting=vec](https://teamcity.cockroachdb.com/viewLog.html?buildId=2870055&tab=artifacts#/sqlsmith/setup=seed-vec/setting=vec)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Asqlsmith%2Fsetup%3Dseed-vec%2Fsetting%3Dvec.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/colexec/external_hash_aggregator.go
+++ b/pkg/sql/colexec/external_hash_aggregator.go
@@ -100,7 +100,7 @@ func NewExternalHashAggregator(
 	// sort isn't accounted for when considering the number file descriptors to
 	// acquire. Not urgent, but it should be fixed.
 	maxNumberActivePartitions := calculateMaxNumberActivePartitions(flowCtx, args, ehaNumRequiredActivePartitions)
-	return createDiskBackedSorter(eha, args.Spec.ResultTypes, outputOrdering.Columns, maxNumberActivePartitions)
+	return createDiskBackedSorter(eha, newAggArgs.OutputTypes, outputOrdering.Columns, maxNumberActivePartitions)
 }
 
 // HashAggregationDiskSpillingEnabled is a cluster setting that allows to

--- a/pkg/sql/logictest/testdata/logic_test/aggregate
+++ b/pkg/sql/logictest/testdata/logic_test/aggregate
@@ -3840,3 +3840,9 @@ SELECT a, b, count(*) FROM t63159 GROUP BY a,b ORDER BY a
 2  2  1
 3  3  1
 5  5  1
+
+# Regression test for the vectorized hash aggregator using incorrect type schema
+# when planning an external sort to maintain the required ordering.
+statement ok
+CREATE TABLE t63436 (a INT, b FLOAT, c DECIMAL, INDEX(a));
+SELECT count(*) FROM t63436@t63436_a_idx GROUP BY b, c ORDER BY c;

